src/lwc/geFormField/__tests__/geFormField.test.js

```diff
@@ -0,0 +1,69 @@
+import {createElement} from 'lwc';
+import GeFormField from 'c/geFormField';
+
+const fieldMapping = require('./data/fieldMapping.json');
+
+jest.mock('c/geFormService', () => {
+    return {
+        getFieldMappingWrapper: jest.fn(() => {
+            return require('./data/fieldMapping.json');
+        }),
+        getInputTypeFromDataType: jest.fn(() => {
+            return true;
+        }),
+        getNumberFormatterByDescribeType: jest.fn(() => {
+            return null;
+        }),
+        getObjectMappingWrapper: jest.fn(() => {
+            return require('./data/objectMapping.json');
+        }),
+        importedRecordFieldNames: jest.fn(() => {
+            return null;
+        }),
+        fieldMappingsForImportedRecordFieldName: jest.fn(() => {
+            return null;
+        })
+    };
+});
+
+describe('c-ge-form-field', () => {
+
+    afterEach(() => {
+        clearDOM();
+        jest.clearAllMocks();
+    });
+
+    it('fires formfieldchange event when change event ' +
+        'from geFormFieldPicklist is handled', () => {
+
+        const elementJSON = require('./data/picklistElement.json');
+
+        const element = createElement('c-ge-form-field', {is: GeFormField});
+        element.element = elementJSON;
+        element.formState = {
+            [fieldMapping.Source_Field_API_Name]: {value: 'testInitialPicklistValue'}
+        };
+
+        const picklistChangeHandler = jest.fn();
+        element.addEventListener('formfieldchange', picklistChangeHandler);
+
+        document.body.appendChild(element);
+
+        expect(element.isPicklist).toBeTruthy();
+
+        const pl = element.shadowRoot.querySelector('lightning-combobox');
+
+        const newPicklistValue = 'newPicklistValue';
+        return Promise.resolve().then(() => {
+            pl.dispatchEvent(new CustomEvent('change', {value: newPicklistValue}));
+        });
+
+        expect(picklistChangeHandler).toBeCalled();
+
+        const formFieldChangeDetailObject = {
+            [fieldMapping.Source_Field_API_Name]: {value: newPicklistValue}
+        }
+        expect(picklistChangeHandler).toBeCalledWith(formFieldChangeDetailObject);
```

Comment:
@kenlewis-sfdo @bdvorachek - It looks to me like this block of code isn't reachable with the `return` on line 57. Could ya'll take another look? Running into a bit of a wall trying to test that the correct events are firing and being handled when a picklist option is selected myself. If ya'll have some time to sync up tomorrow, I'd appreciate it!

Reply:
Yeah, good catch @psadie-sfdo I'll take a look at this test and see if I can re-work it to make it into something meaningful.  Thanks!

Reply:
So I did a bit of research on this and discovered that we need to fast-forward time whenever we're testing this component due to the debounce-ing we're doing in the change handler. We'd put `jest.useFakeTimers();` in the header of this file, and then `jest.runOnlyPendingTimers();` after dispatching the change event.

Reply:
Thanks for looking into this Ben! Gonna give that a shot today!